app/models/submission.rb

```diff
@@ -54,7 +54,7 @@ class Submission < ApplicationRecord
   validate :can_be_ineligible_for_evaluation,
            if: -> { judging_status_change == %w[selected not_selected] }
 
-  after_save :update_submission_evaluation_status
+  before_save :set_evaluation_status
```

Comment:
I'd like to remove this before_save callback to avoid recalculating the status on every save event on the model, when most direct model updates will not impact the status. The service can update the model itself instead.